connect/runtime/src/test/java/org/apache/kafka/connect/integration/MonitorableSinkConnector.java

```diff
@@ -0,0 +1,126 @@
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class MonitorableSinkConnector extends TestSinkConnector {
+
+    public static final String EXPECTED_RECORDS = "expected_records";
+    private static final Logger log = LoggerFactory.getLogger(MonitorableSinkConnector.class);
+
+    public static final Map<String, MonitorableSinkTask> TASKS = new ConcurrentHashMap<>();
+
+    private String connectorName;
+    private String expectedRecordsStr;
+
+    @Override
+    public void start(Map<String, String> props) {
+        connectorName = props.get("name");
+        expectedRecordsStr = props.get(EXPECTED_RECORDS);
+        log.info("Starting connector {}", props.get("name"));
+    }
+
+    @Override
+    public Class<? extends Task> taskClass() {
+        return MonitorableSinkTask.class;
+    }
+
+    @Override
+    public List<Map<String, String>> taskConfigs(int maxTasks) {
+        List<Map<String, String>> configs = new ArrayList<>();
+        for (int i = 0; i < maxTasks; i++) {
+            Map<String, String> config = new HashMap<>();
+            config.put("task.id", connectorName + "-" + i);
+            config.put("required", "dummy-val");
+            config.put(EXPECTED_RECORDS, expectedRecordsStr);
+            configs.add(config);
+        }
+        return configs;
+    }
+
+    @Override
+    public void stop() {
+
+    }
+
+    @Override
+    public ConfigDef config() {
+        return new ConfigDef();
+    }
+
+    public static class MonitorableSinkTask extends SinkTask {
+
+        private String taskId;
+        private int expectedRecords;
+        private CountDownLatch latch;
+
+        @Override
+        public String version() {
+            return "unknown";
+        }
+
+        @Override
+        public void start(Map<String, String> props) {
+            log.debug("Starting task {}", context);
+            taskId = props.get("task.id");
+            expectedRecords = Integer.parseInt(props.get(EXPECTED_RECORDS));
+            TASKS.put(taskId, this);
+            latch = new CountDownLatch(expectedRecords);
+        }
+
+        @Override
+        public void put(Collection<SinkRecord> records) {
+            for (SinkRecord rec : records) {
+                latch.countDown();
+                log.debug("Obtained record: {} at {}", rec.value(), context);
+            }
+        }
+
+        @Override
+        public void stop() {
+            log.info("Removing {}", TASKS.remove(taskId));
+        }
+
+        public void awaitRecords(int consumeMaxDurationMs) throws InterruptedException {
+            if (latch == null) {
+                throw new IllegalStateException("latch was not initialized");
+            } else {
+                if (!latch.await(consumeMaxDurationMs, TimeUnit.MILLISECONDS)) {
+                    throw new DataException("Insufficient records seen by task");
```

Comment:
If I'm not mistaken you know how many records you expect. 
Should we print that to the message ("Fewer than X records were consumed by the task in Y ms") ?